src/Decompiler/Analysis/ConditionCodeEliminator.cs

```diff
@@ -453,30 +471,45 @@ private Instruction TransformRorC(Application rorc, Assignment a)
             if (!(cond.Expression is Identifier condId))
                 return a;
             var sidOrigHi = ssaIds[condId];
-            if (!(sidOrigHi.DefExpression is BinaryExpression shift &&
-                  shift.Operator == Operator.Shr))
+            if (sidOrigHi.DefExpression is Slice slice2)
+            {
+                sidOrigHi = ssaIds[(Identifier) slice2.Expression];
+            }
+            if (!(sidOrigHi.DefExpression is BinaryExpression shift))
+                return a;
+            Domain domain;
+            if (shift.Operator == Operator.Shr)
+                domain = Domain.UnsignedInt;
+            else if (shift.Operator == Operator.Sar)
+                domain = Domain.SignedInt;
+            else 
                 return a;
 
             var block = sidOrigLo.DefStatement!.Block;
             var expShrSrc = shift.Left;
             var expRorSrc = rorc.Arguments[0];
 
-            var stmGrf = sidGrf!.DefStatement;
-            block.Statements.Remove(stmGrf!);
+            foreach (var sid in sidsToKill)
+            {
+                ssa.DeleteStatement(sid.DefStatement!);
+                ssa.Identifiers.Remove(sid);
```

Comment:
What if `sid` has other uses?

Reply:
added a check for >= 2 uses, although these may be very unlikely.